### PR TITLE
Allow customizing app name

### LINKED ISSUE #3220: Allow customizing app name
Currently all the product names ("Joystream" on master/dev and "Atlas" on carthage) are hardcoded into specific components and messages. Let's introduce a variable in the app config letting an operator to easily update the product name to something else

### PR DESCRIPTION
**Pull request by drillprop**
Fix #3220 
FYI - I added a plugin in vite which should read variables in HTML, it was pretty easy to replace

I wasn't sure if I should update atlas-meta-server, but I did this anyway. For some reason, I couldn't read APP_NAME from .env in the  `packages/atlas-meta-server/src/tags.ts` - I was getting `process` is undefined(I guess this is because this code is running on the browser(?)).

### CODE REVIEW

File: packages/atlas-meta-server/src/tags.ts [hunk 1]
```diff
@@ -3,7 +3,8 @@ import { formatISODuration } from 'date-fns'
 import { BasicChannelFieldsFragment, BasicVideoFieldsFragment } from './api/__generated__/sdk'
 import { joinUrlFragments } from './utils'
 
-const BASE_ATLAS_URL = 'https://play.joystream.org'
+const BASE_APP_URL = 'https://play.joystream.org'
+const APP_NAME = 'Atlas'
```
Comment (kdembler) [inline packages/atlas-meta-server/src/tags.ts]: Not sure what the issue you've seen here was but this code doesn't run in the browser, it's all a Node backend. Could you try again to use the env variable here? Feel free to ping me on Slack if it's problematic, I can help

File: packages/atlas-meta-server/src/tags.ts [hunk 2]
```diff
@@ -19,12 +20,12 @@ export type MetaTags = Record<string, string | number>
 
 const commonMetaTags = {
   'twitter:site': '@JoystreamDAO',
```
Comment (kdembler) [inline packages/atlas-meta-server/src/tags.ts]: If we're going through this exercise let's make this configurable via env as well

File: packages/atlas-meta-server/src/tags.ts [hunk 3]
```diff
@@ -3,7 +3,8 @@ import { formatISODuration } from 'date-fns'
 import { BasicChannelFieldsFragment, BasicVideoFieldsFragment } from './api/__generated__/sdk'
 import { joinUrlFragments } from './utils'
 
-const BASE_ATLAS_URL = 'https://play.joystream.org'
+const BASE_APP_URL = 'https://play.joystream.org'
```
Comment (kdembler) [inline packages/atlas-meta-server/src/tags.ts]: If we're going through this exercise let's make this configurable via env as well

**Review by kdembler — CHANGES_REQUESTED** [review]
Other than requested changes, looks good

File: packages/atlas-meta-server/src/config.ts [hunk 4]
```diff
@@ -1,21 +1,13 @@
...
 
-const APP_NAME = process.env.APP_NAME as string
-if (!APP_NAME) {
-  // eslint-disable-next-line no-console
-  console.error('Missing required APP_NAME env variable')
-  process.exit(1)
-}
+const GRAPHQL_URL = getEnvVariable('GRAPHQL_URL', true)
+const APP_NAME = getEnvVariable('APP_NAME', true)
+const TWITTER_ID = getEnvVariable('TWIITER_ID')
```
Comment (kdembler) [inline packages/atlas-meta-server/src/config.ts]: Typo in the string - should be `TWITTER_ID`

Comment (drillprop) [inline packages/atlas-meta-server/src/config.ts]:
```suggestion
const TWITTER_ID = getEnvVariable('TWITTER_ID')
```

File: packages/atlas-meta-server/.env [hunk 5]
```diff
@@ -1,3 +1,5 @@
 # This file is committed. Do not store secrets here
 
 GRAPHQL_URL=https://orion.joystream.org/graphql
+APP_NAME=Atlas
+TWITTER_ID=@JoystreamDAO
```
Comment (drillprop) [inline packages/atlas-meta-server/.env]:
```suggestion
TWITTER_ID=@JoystreamDAO

```

File: packages/atlas/src/config/env.ts [hunk 6]
```diff
@@ -9,6 +9,7 @@ export const AVATAR_SERVICE_URL = readEnv('ASSETS_SERVICE_URL', true, true)
 
 export const SENTRY_DSN = readEnv('SENTRY_DSN', false)
 
+export const TWIITER_ID = readEnv('TWITTER_ID', false, true)
```
Comment (kdembler) [inline packages/atlas/src/config/env.ts]: Typo in the variable name, should be `TWITTER_ID`

File: packages/atlas-meta-server/src/tags.ts (hunk 3 above)
Comment (kdembler) [reply packages/atlas-meta-server/src/tags.ts]: @drillprop This wasn't resolved

**Review by kdembler — APPROVED** [review]

**Pull request merged**